src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -483,6 +559,62 @@ internal FunctionPointerMethodSymbol ApplyNullableTransforms(byte defaultTransfo
             }
         }
 
+        public ImmutableHashSet<CustomModifier> GetCallingConventionModifiers()
+        {
+            if (_lazyCallingConventionModifiers is null)
+            {
+                var modifiersToSearch = RefKind != RefKind.None ? RefCustomModifiers : ReturnTypeWithAnnotations.CustomModifiers;
+                if (modifiersToSearch.IsEmpty || CallingConvention != CallingConvention.Unmanaged)
+                {
+                    Interlocked.CompareExchange(ref _lazyCallingConventionModifiers, ImmutableHashSet<CustomModifier>.Empty, null);
+                }
+                else
+                {
+                    var builder = PooledHashSet<CustomModifier>.GetInstance();
+                    foreach (var modifier in modifiersToSearch)
+                    {
+                        if (IsCallingConventionModifier(modifier))
+                        {
+                            builder.Add(modifier);
+                        }
+                    }
+
+                    if (builder.Count == 0)
+                    {
+                        Interlocked.CompareExchange(ref _lazyCallingConventionModifiers, ImmutableHashSet<CustomModifier>.Empty, null);
+                    }
+                    else
+                    {
+                        Interlocked.CompareExchange(ref _lazyCallingConventionModifiers, builder.ToImmutableHashSet(), null);
+                    }
+
+                    builder.Free();
+                }
+            }
+
+            return _lazyCallingConventionModifiers;
+        }
+
+        private static bool IsCallingConventionModifier(CustomModifier modifier)
```

Comment:
>IsCallingConventionModifier [](start = 28, length = 27)

What is this function going to return for type named "System.Runtime.CompilerServices.CallConv"? Do we have a test for a scenario with a modifier like that?

Reply:
Added `CallingConventionNamedCallConv`

---
In reply to: [457733128](https://github.com/dotnet/roslyn/pull/46110#discussion_r457733128) [](ancestors = 457733128)